### ISSUE TITLE
Cannot deserialize API response

### ISSUE THREAD

**Issue by gatesn** [ory/hydra-client-java#19]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/hydra-client-java/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/hydra-client-java/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Network](https://www.ory.sh/) project.
- [ ] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [X] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

I get an error in the Java client when deserializing a response from the server:

```
Caused by: java.lang.IllegalArgumentException: Expected the field `grant_types` to be an array in the JSON string but got `null`
```

### Reproducing the bug

Making a call to `OAuth2Api#createOAuth2Client`

### Version

2.1.1

### On which operating system are you observing this issue?

macOS

### In which environment are you deploying?

Binary

### Additional Context

The issue is that the OAuth2Api incorrectly checks for null values:

```
      // ensure the optional json data is an array if present
      if (jsonObj.get("grant_types") != null && !jsonObj.get("grant_types").isJsonArray()) {
        throw new IllegalArgumentException(String.format("Expected the field `grant_types` to be an array in the JSON string but got `%s`", jsonObj.get("grant_types").toString()));
      }
```

Instead, it should be:
```
      // ensure the optional json data is an array if present
      if (!jsonObj.get("grant_types").isJsonNull() && !jsonObj.get("grant_types").isJsonArray()) {
        throw new IllegalArgumentException(String.format("Expected the field `grant_types` to be an array in the JSON string but got `%s`", jsonObj.get("grant_types").toString()));
      }
```

**Comment by gatesn** [comment]
Looks like it's fixed upstream here: https://github.com/OpenAPITools/openapi-generator/issues/13548

**Comment by vmuth85** [comment]
I've got the same issue with the hydra-client-v2.2.0-rc.3. I cannot perform the Authorization Code Flow without setting the 'audience' request parameter. The response of 

> GET https://www.ory.sh/admin/oauth2/auth/requests/login?challenge=uJQp89szo......GcAez4qV0-V_ 

looks as follows:

> "challenge": "uJQp89szo......GcAez4qV0-V_",
>     "requested_scope": [
>         "offline_access",
>         "openid"
>     ],
>     "requested_access_token_audience": null,
>     "skip": true,
>     "subject": "user",
>     "oidc_context": {},
> 	...

As you can see, the 'requested_access_token_audience' attribute is null, but the OAuth2LoginRequest.validateJsonObject method doesn't check properly for null values and throws an java.lang.IllegalArgumentException ( "Expected the field `requested_access_token_audience` to be an array in the JSON string but got `null`")

The current code 

```
// ensure the required json array is present
      else if (!jsonObj.get("requested_access_token_audience").isJsonArray()) {
        throw new IllegalArgumentException(String.format("Expected the field `requested_access_token_audience` to be an array in the JSON string but got `%s`", jsonObj.get("requested_access_token_audience").toString()));
      }
```
has to be changed as follows:

>       // ensure the required json array is present
>       else if (!jsonObj.get("requested_access_token_audience").isJsonNull() && !jsonObj.get("requested_access_token_audience").isJsonArray()) {
>         throw new IllegalArgumentException(String.format("Expected the field `requested_access_token_audience` to be an array in the JSON string but got `%s`", jsonObj.get("requested_access_token_audience").toString()));
>       }

**Comment by alnr** [comment]
This same issue was reported for the Rust client in the Community Slack.

**Comment by aeneasr** [comment]
This is fixed now

**Comment by MalteBellmann** [comment]
For me this is still not fixed with server and client versions at 2.2.0. I still get the `IllegalArgumentException` when i call `createOAuth2Client`. In my case the problematic fields are `contacts`, `redirectUris` and `responseTypes`. 

@aeneasr can you take a look at this again?
